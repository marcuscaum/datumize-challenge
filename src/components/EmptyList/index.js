import React from 'react';
import PropTypes from 'prop-types';

import sadEmote from '../../assets/sad_emote.svg';

import EmptyList from './index.styled';

const EmptyListComponent = ({ message }) => (
  <EmptyList>
    <img src={sadEmote} alt="sad_emote" />
    <span>{message}</span>
  </EmptyList>
);

EmptyListComponent.propTypes = {
  message: PropTypes.string.isRequired,
};

export default EmptyListComponent;
