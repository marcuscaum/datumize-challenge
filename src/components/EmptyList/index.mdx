---
name: EmptyList
---


import { Playground, PropsTable } from 'docz'
import EmptyList from './index'

# EmptyList

<PropsTable of={EmptyList} />

## Basic usage

<Playground>
    <EmptyList message="This list is empty"/>
</Playground>