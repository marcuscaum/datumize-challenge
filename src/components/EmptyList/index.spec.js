import React from 'react';
import { shallow } from 'enzyme';

import EmptyList from './index';

const setup = props => shallow(<EmptyList {...props} />);

describe('[Component] EmptyList', () => {
  it('should render the component', () => {
    const wrapper = setup({
      message: 'Empty list',
    });
    expect(wrapper).toMatchSnapshot();
  });
});
