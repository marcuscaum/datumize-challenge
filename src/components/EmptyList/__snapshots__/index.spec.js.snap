// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] EmptyList should render the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EmptyListComponent
    message="Empty list"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <img
          alt="sad_emote"
          src="sad_emote.svg"
/>,
        <span>
          Empty list
</span>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "sad_emote",
          "src": "sad_emote.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Empty list",
        },
        "ref": null,
        "rendered": "Empty list",
        "type": "span",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <img
            alt="sad_emote"
            src="sad_emote.svg"
/>,
          <span>
            Empty list
</span>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "sad_emote",
            "src": "sad_emote.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Empty list",
          },
          "ref": null,
          "rendered": "Empty list",
          "type": "span",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
